### PR TITLE
Mat 4489

### PR DESCRIPTION
**Pull request by ethankaplan**
## MADiE PR

Jira Ticket: [MAT-4489](https://jira.cms.gov/browse/MAT-4489)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [x] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

**Review by gregory-akins — DISMISSED** [review]

File: src/components/measureGroups/MeasureGroups.tsx [hunk 1]
```diff
@@ -244,6 +270,12 @@ const MeasureGroups = () => {
   };
 
   const submitForm = (group: Group) => {
+    if (group.stratifications) {
+      group.stratifications = group.stratifications.filter(
+        (strat) => !!strat.description || !!strat.cqlDefinition
```
Comment (adongare) [inline src/components/measureGroups/MeasureGroups.tsx:280]: what is the benefit of using double negation here?

File: src/components/measureGroups/MeasureGroups.tsx [hunk 2]
```diff
@@ -517,7 +554,34 @@ const MeasureGroups = () => {
                   <MenuItem
                     data-testid="stratifications-tab"
                     isActive={activeTab == "stratification"}
-                    onClick={() => setActiveTab("stratification")}
+                    onClick={() => {
+                      setActiveTab("stratification");
+                      if (!!formik.values.stratifications) {
```
Comment (adongare) [inline src/components/measureGroups/MeasureGroups.tsx:566]: same question here as well(double negation)

File: src/components/measureGroups/MeasureGroups.tsx [hunk 3]
```diff
@@ -517,7 +554,34 @@ const MeasureGroups = () => {
...
+                          formik.values.stratifications.push({
+                            cqlDefinition: "",
+                            description: "",
+                            association: "",
+                            id: "",
+                          });
+                        }
+                      } else {
+                        formik.values.stratifications = [
+                          {
```
Comment (adongare) [inline src/components/measureGroups/MeasureGroups.tsx:574]: seems like we can avoid duplication of this object at multiple places. we can define this object at one place and use the spread operator to copy wherever needed

File: src/components/measureGroups/MeasureGroups.tsx [hunk 4]
```diff
@@ -577,8 +641,114 @@ const MeasureGroups = () => {
             )}
             {activeTab === "stratification" && (
               <FormControl>
-                <Divider />
-                <FieldLabel>Stratification support to come</FieldLabel>
+                {formik.values.stratifications.map((strat, i) => (
+                  <Row>
```
Comment (adongare) [inline src/components/measureGroups/MeasureGroups.tsx:644]: Fine for now but feels like this whole row can be a separate component?

File: src/components/measureGroups/MeasureGroups.tsx (hunk 1 above)
Comment (ethankaplan) [reply src/components/measureGroups/MeasureGroups.tsx:280]: In typescript !! means truthiness of the object.

File: src/components/measureGroups/MeasureGroups.tsx (hunk 3 above)
Comment (ethankaplan) [reply src/components/measureGroups/MeasureGroups.tsx:574]: Fixed, thank you

File: src/components/measureGroups/MeasureGroups.tsx (hunk 4 above)
Comment (ethankaplan) [reply src/components/measureGroups/MeasureGroups.tsx:644]: Added a comment

**Review by gregory-akins — APPROVED** [review]

**Review by adongare — APPROVED** [review]
